### PR TITLE
BoozerLS surface optimization

### PR DESCRIPTION
**Pull request by andrewgiuliani**
This PR introduces to SIMSOPT the BoozerLS optimization formulation described in

Giuliani, A., Wechsung, F., Cerfon, A., Landreman, M., & Stadler, G. (2023). Direct stellarator coil optimization for nested magnetic surfaces with precise quasi-symmetry. Physics of Plasmas, 30(4).

To dos:
- [x] unit tests
- [x] non-XSIMD implementation of BoozerLS objective evaluation
- [x] docstrings

### CODE REVIEW

**Comment by mbkumar** [comment]
Hello Andrew,

Is this branch abandoned?

**Comment by andrewgiuliani** [comment]
Hi Bharat, I still have to work on the items in the to do list, so it's not abandoned. Thanks for checking in.

File: src/simsopt/geo/boozersurface.py [hunk 1]
```diff
@@ -40,17 +40,48 @@ class BoozerSurface(Optimizable):
...
         self.surface = surface
         self.label = label
         self.targetlabel = targetlabel
+        self.constraint_weight = constraint_weight
         self.need_to_run_code = True
 
     def recompute_bell(self, parent=None):
         self.need_to_run_code = True
+    
+    def run_code(self, boozer_type, iota, G=None, verbose=True):
```
Comment (ejpaul) [inline src/simsopt/geo/boozersurface.py:56]: I suggest making tol and maxiter keyword arguments (with defaults) so the user can adjust if necessary.

File: src/simsopt/geo/boozersurface.py
Comment (ejpaul) [inline src/simsopt/geo/boozersurface.py:1]: I suggest adding some text to the docstring about requirements for quadpoints in LS vs. exact

File: src/simsopt/geo/boozersurface.py (hunk 1 above)
Comment (andrewgiuliani) [reply src/simsopt/geo/boozersurface.py]: I've added support for solver options now, thanks!

**Pull request merged**